scylla-cdc-base/src/main/java/com/scylladb/cdc/model/worker/Worker.java

```diff
@@ -100,32 +129,32 @@ private TaskActionsQueue queueFirstActionForEachTask(Map<TaskId, SortedSet<Strea
      * At each iteration, runs a single action from |actions| queue if any
      * available.
      */
-    private void performActionsUntilStopRequested(TaskActionsQueue actions) {
-        while (!workerConfiguration.transport.shouldStop()) {
-            try {
-                actions.runNextAction();
-            } catch (InterruptedException e) {
-                // Ignore InterruptedException
-            }
+    private void performActionsUntilStopRequested(Collection<TaskAction> actions) {
+        if (shouldStop()) {
+            return;
         }
 
-        // Stop all "sleeping" futures.
+        ScheduledExecutorService executorService = getExecutorService();
         try {
-            workerConfiguration.delayedFutureService.stop();
+            executorService.invokeAll(actions.stream().map(a -> makeCallable(a)).collect(Collectors.toSet()));
+            do {
+                executorService.awaitTermination(10, MINUTES);
+            } while (!shouldStop() && !executorService.isTerminated());
         } catch (InterruptedException e) {
-            // Ignore InterruptedException
+            logger.atWarning().log("Worker interrupted");
         }
-
-        // Wait for already started actions to gracefully
-        // finish.
-        actions.join();
     }
 
+    public void stop() {
+        shouldStop = true;
+        getExecutorService().shutdown();
```

Comment:
Should it be `shutdownNow()`? The way it is now it introduces a regression in time to stop the Worker (I think it waits up to query window size time - 30 seconds).

Why is it important: 
1. Better user experience of using example applications
2. No warninigs/errors while shutting down Kafka connector - I believe that Kafka Connect has some timeout for closing connector and this could trigger it
3. Faster switching of generations (in Kafka connector) - when switching generations we wait for all Workers to stop. The faster this process is, the smaller the lag will be.

Reply:
Well, I noticed I forgot the kafka connector (weird - why was it not red before?) more or less needs the polling shutdown to work too, so need to retain it. 
But as for shutdown - I think this should be up to user. shutdownNow is not great, because it interrupts executing tasks. 
But a user (and we should in default) can set `ScheduledExecutor::setExecuteExistingDelayedTasksAfterShutdownPolicy(false)` to ensure that shutdown does not wait for scheduled "readnextwindow" etc.

Reply:
I changed the default executor to not wait for scheduled tasks. Calling stop and reaching end-of-process is now instantaneous. Also reverted removing the WorkerTransport.shouldStop, so app can use polled shutdown as well. Reduced sleep delay in that poll to 50ms (as before), so should also be quick.

Reply:
Please add a comment in the code so that this discussion is persisted for the future.

Reply:
I will add it in javadoc for the builder.